Sources/StructuredQueriesMacros/SelectionMacro.swift

```diff
@@ -98,6 +98,9 @@ extension SelectionMacro: ExtensionMacro {
             columnQueryValueType = "\(raw: base.trimmedDescription)"
 
           case .some(let label) where label.text == "primaryKey":
+            guard !declaration.hasMacroApplication("Table")
+            else { continue }
```

Comment:
Unrelated to the core of this PR but something I noticed. We can allow `@Selection` types to have a primary key if it is also a `@Table`.